consensus/raft/raft.go

```diff
@@ -686,3 +695,47 @@ func find(s []string, elem string) bool {
 	}
 	return false
 }
+
+func (rw *raftWrapper) observePeers() {
+	obsCh := make(chan hraft.Observation, 1)
+	defer close(obsCh)
+
+	if !sixtyfour {
+		// 32-bit systems don't support observers
+		// TODO: Check if this is still the case?
+		logger.Warning("raft Observers panic on 32-bit architecture")
+		return
+	}
+
+	observer := hraft.NewObserver(obsCh, false, nil)
```

Comment:
why not blocking?

Reply:
and why not filtering?

Reply:
filtering I have done after receiving observations, but I can change it to here.

Making blocking true will cause raft to block when sending observation. I dont see a strong reason to do so. Generally it should be false.
https://github.com/hashicorp/raft/blob/ff523e1c1515a5b33cc3153a34c984093fed06bb/observer.go#L50-L51

Reply:
> I dont see a strong reason to do so

Think again. If several peers are removed at once and Raft cannot put them in the channel, you will never get notified. Yes, there are strong reasons to block so you don't miss any observation. There are mild reasons to filter. I'm ok with no filtering.